### PR TITLE
add documentation for response models

### PR DESCRIPTION
**Pull request by LizaShak**
This PR resolved the following task: [1361](https://dev.azure.com/csedevil/K2-bridge-internal/_workitems/edit/1361)

The following changes are proposed:

Add documentation to response models

### CODE REVIEW

File: K2Bridge/Models/Response/Metadata/FieldCapabilityResponse.cs [hunk 1]
```diff
@@ -7,13 +7,24 @@ namespace K2Bridge.Models.Response.Metadata
...
+    /// <summary>
+    /// Field capability response.
+    /// </summary>
     public class FieldCapabilityResponse
     {
         private readonly Dictionary<string, FieldCapabilityElement> fields = new Dictionary<string, FieldCapabilityElement>();
 
+        /// <summary>
+        /// Gets all fields dictionary.
+        /// Where key is the field name and the value is the field capability element.
```
Comment (boazsha) [inline K2Bridge/Models/Response/Metadata/FieldCapabilityResponse.cs]: I would replace the word "where" (especially after a full-stop)

File: K2Bridge/Models/Response/Metadata/IndexListResponseElement.cs [hunk 2]
```diff
@@ -6,8 +6,14 @@ namespace K2Bridge.Models.Response.Metadata
 {
     using Newtonsoft.Json;
 
+    /// <summary>
+    /// Indices collecton response element.
```
Comment (boazsha) [inline K2Bridge/Models/Response/Metadata/IndexListResponseElement.cs]: Index (single and not plural)

File: K2Bridge/Models/Response/Shards.cs [hunk 3]
```diff
@@ -6,17 +6,38 @@ namespace K2Bridge.Models.Response
...
 
+    /// <summary>
+    /// Object containing a count of shards used for the request.
+    /// </summary>
     public class Shards
     {
+        /// <summary>
+        /// Gets or sets the total number of shards.
+        /// It is here only to support the expected response.
+        /// And In our case it is a const == 1.
```
Comment (boazsha) [inline K2Bridge/Models/Response/Shards.cs]: Proposal: "Exists in order to support Kibana's expected response".

Also I would discard the word "And".

(Relevant for the following properties as well)

File: K2Bridge/Models/Response/Shards.cs [hunk 4]
```diff
@@ -6,17 +6,38 @@ namespace K2Bridge.Models.Response
...
     using Newtonsoft.Json;
 
+    /// <summary>
+    /// Object containing a count of shards used for the request.
+    /// </summary>
     public class Shards
     {
+        /// <summary>
+        /// Gets or sets the total number of shards.
+        /// Exists in order to support Kibana's expected response
```
Comment (eladiw) [inline K2Bridge/Models/Response/Shards.cs]:
```suggestion
        /// Exists in order to support Kibana's expected response (default: 1)
```

Comment (eladiw) [reply K2Bridge/Models/Response/Shards.cs]: remove below line

(same for others)

**Pull request merged**